csm_web/scheduler/views/section.py

```diff
@@ -17,19 +19,23 @@
     Spacetime,
     Student,
     User,
+    Swap,
 )
 from scheduler.serializers import (
     SectionOccurrenceSerializer,
     SectionSerializer,
     SpacetimeSerializer,
     StudentSerializer,
+    SwapSerializer,
 )
 
 from .utils import get_object_or_error, log_str, logger, viewset_with
 
 
 class SectionViewSet(*viewset_with("retrieve", "partial_update", "create")):
     serializer_class = SectionSerializer
+    lookup_field = 'pk'
+    lookup_url_kwarg = 'section_id'
```

Comment:
These seem to break the existing section tests due to the change in URL matching. I'm unfamiliar with these viewset fields, but this should be looked into; either modify this, or correct the existing tests (and potentially serializers?) that may rely on the URL match groups.

Reply:
@edwardneo I remember this coming from a commit of yours for the swap DELETE method. I have removed this for now since it's messing up the URL matcher for all the other tests and I was able to get the swap tests to work without this by making a few changes in the parameter names for the `swap_with_id` and `swap_no_id` functions. Lmk if you need me to add this back for any code extensions you plan to make in the future. Leaving this unresolved for now.